src/App.tsx

```diff
@@ -15,37 +17,97 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+export const BY_ALPHABETICALLY = 'alphabetically';
+export const BY_LENGTH = 'length';
+export const BY_REVERSE = 'reverse';
+
+type SortField = typeof BY_ALPHABETICALLY | typeof BY_LENGTH | null;
+
+export function getSortList(
+  goods: string[],
+  sortField: SortField,
+  reverse: boolean,
+): string[] {
+  const sortGoods = [...goods];
+
+  switch (sortField) {
+    case BY_ALPHABETICALLY:
+      sortGoods.sort();
+      break;
+    case BY_LENGTH:
+      sortGoods.sort((good1, good2) => good1.length - good2.length);
+      break;
+    default:
+      break;
+  }
+
+  if (reverse) {
+    sortGoods.reverse();
+  }
+
+  return sortGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortField>(null);
+  const [reverse, setReverse] = useState<boolean>(false);
+
+  const handleSort = (field: SortField | typeof BY_REVERSE) => {
+    if (field === BY_REVERSE) {
+      setReverse(prevReverse => !prevReverse);
+    } else {
+      setSortField(field);
+    }
+  };
+
+  const handleReset = () => {
+    setSortField(null);
+    setReverse(false);
+  };
```

Comment:
The function's name 'handleReset' should start with a verb that clearly indicates its action. A better name would be 'resetSorting', which clearly describes what the function does, following the [CODE STYLE] checklist item.